### ISSUE TITLE
Add support for cloth simulation

### ISSUE THREAD

**Issue by xPaw** [ValveResourceFormat/ValveResourceFormat#209]
For example gman animates, but not properly.

**Comment by CosmicDreamsOfCode** [comment]
I think the animations used for gman are made for "gman_split"
They work fine with that model

**Comment by xPaw** [comment]
Interesting. There's certainly some models that get bugged animations (not just characters).

And there's logs like these:
```
[17:10:43.405] Unhandled animation bone decoder type 'CCompressedStaticQuaternion'
[17:10:43.406] Unhandled animation bone decoder type 'Unknown'
```

**Comment by lewa-j** [comment]
It looks like the broken part are mostly clothes. Maybe it's related to cloth sim?

**Comment by Perryvw** [comment]
Results of some investigation, there are seemingly multiple issues:

1) Cloth sim indeed seems part of the problem. The model I saw is broken because the cloth bones have no parents set, and thus stretch to (0,0,0) instead of following the bone they are attached to. I could not find where the cloth bones are defined to attach or at least follow existing bones (which would also be required for cloth simulation).
2) Bodies seem not animated. I am not sure what the reason for this is, but my **guess** is that valve is animating the body at runtime using Inverse Kinematics. Not sure how to confirm this. I almost hope this is not the case because that is complicated to implement.
3) Some bone rotations seem to be around the wrong axis. My guess here is we are either forgetting  some 'base' transformation of these bones, or we are incorrectly decoding some quaternions.

**Comment by xPaw** [comment]
I looked at it in model doc and blender, one of the problems appear to be related to mesh groups (combine_captain, head doesn't  animate correctly in vrf but does in blender).

And I looked at these character models in valve's modeldoc, and there's similar stretching bugs to what I believe is default bind position (same in blender, but vrf stretches to 0,0 and loses scale?).

In modeldoc stretching only happens when toggling physics preview, it does not appear to be related to cloth simulation (it looks like gman's entire body is not even clothes, just the belly).

**Comment by kristiker** [comment]
Before
https://github.com/ValveResourceFormat/ValveResourceFormat/assets/26466974/d2890f2c-c1b3-4a39-b6e6-8aaf7cbed4af

And after commit: a6657055e43b2c88ac6e7a056f1e94a4e9f529a1 `Use ivec4 bone indices`


https://github.com/ValveResourceFormat/ValveResourceFormat/assets/26466974/5134902c-720b-4ff3-a0e4-2d1ccb57c796

**Comment by kristiker** [comment]
This last issue is related to cloth simulation, so I'm changing the title.

**Comment by xPaw** [comment]
Fixed by ca8793e830e099418158e0084aa9c6721f54434d.

> Copy cloth node transforms from one of the root bones that also contains the Cloth flag, but is not procedural and is driven by animation.
